nifi-commons/nifi-record/src/main/java/org/apache/nifi/serialization/record/util/DataTypeUtils.java

```diff
@@ -1378,6 +1393,61 @@ public static boolean isCharacterTypeCompatible(final Object value) {
         return value != null && (value instanceof Character || (value instanceof CharSequence && ((CharSequence) value).length() > 0));
     }
 
+    public static BigDecimal toDecimal(final Object value, final String fieldName) {
+        if (value == null) {
+            return null;
+        }
+
+        if (value instanceof BigDecimal) {
+            return (BigDecimal)value;
+        }
+
+        if (value instanceof String) {
+            return new BigDecimal((String) value);
+        }
+
+        if(value instanceof Number){
+            return new BigDecimal(((Number)value).doubleValue());
+        }
+
```

Comment:
I would also check for precision here and throw an `IllegalTypeConversionException` if the required precision is less than the precision of the converted `BigDecimal`.

Reply:
Good feedback, will add that